pkgs/development/tools/phase/default.nix

```
@@ -0,0 +1,98 @@
+{ lib, stdenv, fetchurl, unzip, installShellFiles, makeWrapper, autoPatchelfHook, zlib, glibc}:
+
+let
+  pname = "phase";
+  version = "1.18.6"; # This will be auto-updated by update.sh
+  hashes = builtins.fromJSON (builtins.readFile ./hashes.json);
+
+  # Binary target map
+  systemInfo = {
+    "x86_64-linux" = {
+      archivePath = "Linux-binary";
+      archString = "linux_amd64";
+    };
+    "aarch64-linux" = {
+      archivePath = "Linux-binary-arm64";
+      archString = "linux_arm64";
+    };
+    "x86_64-darwin" = {
+      archivePath = "Darwin-binary";
+      archString = "macos_amd64";
+    };
+    "aarch64-darwin" = {
+      archivePath = "Darwin-binary-arm64";
+      archString = "macos_arm64";
+    };
+  };
+
+  system = stdenv.hostPlatform.system;
+  info = systemInfo.${system} or (throw "Unsupported system: ${system}");
+
+  runtimeDeps = [
+    stdenv.cc.cc.lib
+    zlib
+    glibc
+  ];
+in
+stdenv.mkDerivation {
+  inherit pname version;
+
+  src = fetchurl {
+    url = "https://github.com/phasehq/cli/releases/download/v${version}/phase_cli_${info.archString}_${version}.zip";
+    hash = hashes.${system};
+  };
+
+  sourceRoot = ".";
+  
+  nativeBuildInputs = [
+    unzip
+    installShellFiles
+    makeWrapper
+    autoPatchelfHook  # This will automatically fix library references
+  ];
+
+  buildInputs = runtimeDeps;
+
+  unpackPhase = ''
+    runHook preUnpack
+    unzip $src
+    runHook postUnpack
+  '';
+
+  installPhase = ''
+    runHook preInstall
+
+    # Create a directory to hold both the binary and _internal
+    mkdir -p $out/libexec/phase
+    
+    # Install both the binary and _internal directory to libexec
+    install -Dm755 "${info.archivePath}/phase/phase" "$out/libexec/phase/phase"
+    cp -r "${info.archivePath}/phase/_internal" "$out/libexec/phase/_internal"
+    
+    # Create the bin directory
+    mkdir -p $out/bin
+    
+    # Create a wrapper script that ensures the binary can find _internal
+    makeWrapper $out/libexec/phase/phase $out/bin/phase \
+      --prefix LD_LIBRARY_PATH : ${lib.makeLibraryPath runtimeDeps} \
+      --set PYTHONPATH $out/libexec/phase \
+      --set PYTHONHOME $out/libexec/phase
+
+    runHook postInstall
+  '';
+
+  # Prevent stripping of binaries as it might break PyInstaller's binary
+  dontStrip = true;
+
+  meta = with lib; {
```

```suggestion
  meta = {
```
https://github.com/NixOS/nixpkgs/issues/208242